js/src/Figure.ts

```diff
@@ -256,8 +256,10 @@ class Figure extends widgets.DOMWidgetView {
 
             // In the classic notebook, we should relayout the figure on
             // resize of the main window.
-            window.addEventListener('resize', () => {
-                this.relayout();
+            const relayoutMethod = this.relayout.bind(this);
+            window.addEventListener('resize', relayoutMethod);
+            this.once('remove', () => {
```

Comment:
You can also re-implement the `remove` method. `window.addEventListener('resize', this.relayout, this);` and `window.removeEventListener('resize', this.relayout, this);` when removing. We might want to remove other event listeners in this method.

Reply:
I prefer it this way, now the code is more local to each other.